rust/hyperlane-base/src/contract_sync/mod.rs

```diff
@@ -27,100 +31,199 @@ const SLEEP_DURATION: Duration = Duration::from_secs(5);
 /// Entity that drives the syncing of an agent's db with on-chain data.
 /// Extracts chain-specific data (emitted checkpoints, messages, etc) from an
 /// `indexer` and fills the agent's db with this data.
-#[derive(Debug, new, Clone)]
-pub struct ContractSync<T, D: HyperlaneLogStore<T>, I: Indexer<T>> {
+#[derive(Debug)]
+pub struct ContractSync<T: Indexable, D: HyperlaneLogStore<T>, I: Indexer<T>> {
     domain: HyperlaneDomain,
     db: D,
     indexer: I,
     metrics: ContractSyncMetrics,
+    broadcast_sender: Option<BroadcastSender<H512>>,
     _phantom: PhantomData<T>,
 }
 
+impl<T: Indexable, D: HyperlaneLogStore<T>, I: Indexer<T>> ContractSync<T, D, I> {
+    /// Create a new ContractSync
+    pub fn new(domain: HyperlaneDomain, db: D, indexer: I, metrics: ContractSyncMetrics) -> Self {
+        Self {
+            domain,
+            db,
+            indexer,
+            metrics,
+            broadcast_sender: T::broadcast_channel_size().map(BroadcastSender::new),
+            _phantom: PhantomData,
+        }
+    }
+}
+
 impl<T, D, I> ContractSync<T, D, I>
 where
+    T: Indexable + Debug + Send + Sync + Clone + Eq + Hash + 'static,
     D: HyperlaneLogStore<T>,
     I: Indexer<T> + 'static,
 {
     /// The domain that this ContractSync is running on
     pub fn domain(&self) -> &HyperlaneDomain {
         &self.domain
     }
-}
 
-impl<T, D, I> ContractSync<T, D, I>
-where
-    T: Debug + Send + Sync + Clone + Eq + Hash + 'static,
-    D: HyperlaneLogStore<T>,
-    I: Indexer<T> + 'static,
-{
+    fn get_new_receive_tx_channel(&self) -> Option<BroadcastReceiver<H512>> {
+        // Create a new channel if it doesn't exist
```

Comment:
nit: is this accurate? made me think that a new broadcast channel was created altogether, but i guess it's just a new receiver end - led me to misunderstand how it works for a few mins

Reply:
I definitely didn't realize tx meant transmitter 😅
Bc we use tx in many places to mean transaction might be nice to keep tx for transaction in the future